sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -40,11 +40,11 @@ namespace Azure { namespace Core { namespace Http {
   struct CurlTransportOptions final
   {
     /**
-     * @brief Default Maximum time in seconds that you allow the connection phase to the server to
-     * take.
+     * @brief Default Maximum time in milliseconds that you allow the connection phase to the server
+     * to take.
      *
      */
-    AZ_CORE_DLLEXPORT static const long DefaultConnectionTimeout = 300;
+    AZ_CORE_DLLEXPORT static const long DefaultConnectionTimeout = 300000;
```

Comment:
Now that I think about this some more, changing the semantics of this option is a behavioral breaking change, which might be surprising to existing users. What's the benefit here? @JeffreyRichter  what do you think?

Have we tried exploring alternatives that might be non-breaking or is this case one where it is worth it to take the break?

Reply:
Nevermind, based on discussion with Victor offline, this change hasn't GA'd yet, so not a breaking change.

https://github.com/Azure/azure-sdk-for-cpp/blob/14a4459ae02ae377f9da0d6ca272291ffb77f4d5/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp#L40

Reply:
Why not `std::chrono::milliseconds`?

Reply:
Does this constant really need to be public? Would the customer ever reference it? If not, move it to `_detail` for now, outside of this struct?

I can see them using and modifying the non-const `ConnectionTimeout` option below, but don't see them referencing this const value.

Reply:
The public header would be using a private (_detail) header?  That would make the _detail to be kept as non-breaking anyway.

Reply:
We have similar pattern elsewhere, like:
https://github.com/Azure/azure-sdk-for-cpp/blob/b8013185d07a52e44a341b85b3a15ba6274c3676/sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_options.hpp#L76

https://github.com/Azure/azure-sdk-for-cpp/blob/b8013185d07a52e44a341b85b3a15ba6274c3676/sdk/storage/azure-storage-queues/inc/azure/storage/queues/protocol/queue_rest_client.hpp#L494-L496

I am not suggesting putting it in a separate header, just the `_detail` namespace and not be part of the struct.

The user will know not to reference anything with `_detail` and we are allowed to then rename or move this constant (per documented contract, not a breaking change). More importantly, this hides the constant a bit since the user doesn't and shouldn't care about it.

If you look at the options today, it seems like the default timeout is the thing the user needs to modify, but it isn't.
Unless you see a good reason for this to be public and as part of the options struct, I recommend "hiding" it.

Reply:
moved it to details